src/NServiceBus.Metrics/Probes/DurationProbeBuilder.cs

```diff
@@ -0,0 +1,42 @@
+namespace NServiceBus.Metrics
+{
+    using System;
+    using System.Reflection;
+    using Features;
+
+    /// <summary>
+    /// Defines a custom metric.
+    /// </summary>
+    abstract class DurationProbeBuilder
+    {
+        protected abstract void WireUp(FeatureConfigurationContext context, DurationProbe probe);
+
+        public DurationProbe Build(FeatureConfigurationContext context)
+        {
+            var probe = GetProbe();
+
+            WireUp(context, probe);
+
+            return probe;
+        }
+
+        DurationProbe GetProbe()
+        {
+            var attr = GetType().GetCustomAttribute<ProbePropertiesAttribute>();
```

Comment:
This requires a null check, and throwing a user friendly exception like below, if not attribute found.

`$The type '{GetType()}' is not annotated with 'ProbePropertiesAttribute'. Add this attribute for providing probe's metadata`